### PR TITLE
Adjusting to new xacros

### PR DESCRIPTION
**Pull request by mikramarc**
## Proposed changes

Adjusting autodetector to pass in new flags to work with new xacros. Some additional cleanup.

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added automated tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [x] Manually tested on hardware (if hardware specific or related).

## Documentation

- [x] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

**Review by beatrizleon — APPROVED** [review]

File: include/sr_hand_detector/sr_hand_detector.h [hunk 1]
```diff
@@ -30,26 +30,26 @@ namespace sr_hand_detector
...
+    void get_available_port_names();
+    void detect_hand_ports();
+    void get_hands_ports_and_serials();
+    int count_slaves_on_port(std::string);
+    int get_hand_serial(std::string);
+    int read_eeprom(int, int, int);
+
+    std::map<int, std::string> hand_serial_to_port_map_;
+    std::vector<std::string> available_port_names_;
+    uint8 ebuf_[MAXBUF];
```
Comment (giusebar) [inline include/sr_hand_detector/sr_hand_detector.h:49]: why not setting maxbuf as const as well?

File: src/sr_hand_detector/sr_hand_autodetect.cpp [hunk 2]
```diff
@@ -76,98 +90,176 @@ void SrHandAutodetect::detect_hands()
...
+
+  throw std::runtime_error("Unknown hand type!");
 }
 
-void SrHandAutodetect::run()
+std::string SrHandAutodetect::vector_to_xacro_string(const std::vector<std::string> &vec)
 {
-  detect_hands();
-  compose_command_suffix();
+  std::string result;
```
Comment (giusebar) [inline src/sr_hand_detector/sr_hand_autodetect.cpp:216]: we should check if vec size is 0 and return None?

File: src/sr_hand_detector/sr_hand_autodetect.cpp [hunk 3]
```diff
@@ -76,98 +90,176 @@ void SrHandAutodetect::detect_hands()
...
+
+std::string SrHandAutodetect::command_suffix_bimanual_per_hand(const YAML::Node &hand_info)
+{
+  std::string mapping_path_suffix_component;
+  std::string side = hand_info["side"].as<std::string>();
 
-    command_suffix_ += " eth_port:=" + rh_eth_port + "_" + lh_eth_port + " rh_serial:=" +
-      std::to_string(rh_serial) + " lh_serial:=" + std::to_string(lh_serial) +
-      " hand_type:=" + rh_hand_type + mapping_path_suffix_component;
+  if (hand_info["mapping_path"])
```
Comment (giusebar) [inline src/sr_hand_detector/sr_hand_autodetect.cpp:179]: what happens if there is no mapping path?

File: src/sr_hand_detector/sr_hand_autodetect.cpp (hunk 2 above)
Comment (giusebar) [reply src/sr_hand_detector/sr_hand_autodetect.cpp:216]: just assuming there might be an extreme case in which people are able to lunch a hand with no fingers (e.g. phantom). if that is not possible in the xacro launch then should be ok.

**Review by giusebar — CHANGES_REQUESTED** [review]

**Review by giusebar — APPROVED** [review]

**Review by biofotis — APPROVED** [review]

**Pull request merged**